backend-lib/src/main/java/cash/z/ecc/android/sdk/internal/model/JniTransactionDataRequest.kt

```diff
@@ -0,0 +1,38 @@
+package cash.z.ecc.android.sdk.internal.model
+
+import androidx.annotation.Keep
+import cash.z.ecc.android.sdk.internal.ext.isInUIntRange
+
+/**
+ * Serves as cross layer (Kotlin, Rust) communication class.
+ */
+@Keep
+interface JniTransactionDataRequest {
```

Comment:
I'm unsure if we can pass such a construct from FFI via JNI to Kotlin. Should we rather switch to a standard class as we do in the case of the other `Jni` classes? When I run the current logic. I see this error:
```
CompactBlockProcessorException$EnhanceTransactionError$EnhanceTxDataRequestsError: Error while attempting to request transactions data from the Rust backend
Caused by: java.lang.ClassNotFoundException: Didn't find class "cash.z.ecc.android.sdk.internal.model.JniTransactionDataRequest.GetStatus"
```

Reply:
Fixed by switching to `sealed class` on the Kotlin side and by changing the path to the nested classes on the Rust FFI side.

Reply:
Yep, I expected this part to fail in this way (I've never used subclasses of interfaces this way, but couldn't get `sealed class` to work myself).